migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,300 @@
+		}
+		defer func() { _ = rows.Close() }()
+		var convertedReportConfigs []*updatedSchema.ReportConfigurations
+		var reportSnapshots []*updatedSchema.ReportSnapshots
+		var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+		for rows.Next() {
+			// convert to report config proto
+			var reportConfig *updatedSchema.ReportConfigurations
+			if err = query.ScanRows(rows, &reportConfig); err != nil {
+				return errors.Wrap(err, "failed to scan rows")
+			}
+			reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert report config from gorm to proto %+v ", reportConfig)
+			}
+			if reportConfigProto.Version == 2 {
+				continue
+			}
+			// skip if version=0 and scope id is not nil since it is v2 config created in tech preview
+			if reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil {
+				reportConfigProto.Version = 2
+				// convert report config proto back to gorm model
+				convertedGormConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+				if err == nil {
+					convertedReportConfigs = append(convertedReportConfigs, convertedGormConfig)
+				} else {
+
+					log.Errorf("failed to convert report config from proto %+v with error %+v", reportConfigProto, err)
+				}
+
+				continue
+			}
+
+			//since checkifNotifierExsists only reads data from older migration, no need to write a new tx
+			notifierFound, err := checkifNotifierExsists(reportConfigProto.GetEmailConfig().GetNotifierId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query notifier with id %s", reportConfigProto.GetEmailConfig().GetNotifierId())
+			}
+
+			if !notifierFound {
+				log.Errorf("Did not find notifier with id %s attached to report config %+v", reportConfigProto.GetEmailConfig().GetNotifierId(),
+					reportConfigProto)
+				continue
+			}
+
+			// create v2 report config from v1
+			newConfig := createV2reportConfig(reportConfigProto)
+
+			// create notifier config for report_configuration_notifier table
+			notifierConfig := createNotifier(reportConfigProto)
+			newConfig.Notifiers = append(newConfig.Notifiers, notifierConfig)
+			// assign version to 2 to new copy and version to 1 in original so that they are not re-created during migration
+			newConfig.Version = 2
+			reportConfigProto.Version = 1
+
+			//if deterministic id exists no need to copy the config
+			//since getMigratedReportConfigIfExsists only reads data from older migration, no need to write a new tx
+			migrated, data, err := getMigratedReportConfigIfExsists(newConfig.GetId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query report config with id %s", reportConfigProto.GetId())
+			}
+			if migrated {
+				if !reflect.DeepEqual(newConfig, data) {
+					log.Errorf("Old v1 report config proto %+v is different from v2 copy", data)
```

Comment:
```suggestion
					log.Errorf("v1 report config %+v was migrated to create v2 report config with id '%s'.  The v1 config has changed since last migration. Skipping re-migration...", reportConfigProto)
```